packages/root/src/Root.spec.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { render, cleanup } from '@testing-library/react';
+import Root from '.';
+
+afterAll(cleanup);
+
+describe('packages/root', () => {
+  describe('by default', () => {
+    const props = { name: 'testName' };
+    const content = 'button content';
+    const { getByTestId } = render(
+      <Root data-testid="button" {...props}>
+        {content}
+      </Root>,
+    );
+    const button = getByTestId('button');
+
+    test('it renders as a button when no props are set', () => {
+      expect(button.tagName.toLowerCase()).toBe('button');
+    });
+    test('it renders children inside of <Root /> tag', () => {
+      expect(button.textContent).toBe(content);
+    });
+
+    test('it renders other props passed, when they are set', () => {
+      expect((button as HTMLButtonElement).name).toBe('testName');
+    });
+  });
+
+  describe('when the href prop is set', () => {
```

Comment:
You've established a nice pattern here in the test of rendering and checking the component, the tag, and the expected prop handling. Since you use it three times here *(and since we'll likely have to expand the tests if we ever have other components we explicitly check for), you could abstract this to a list of the data to test and then map over it to run your tests.

For example, here's a good chunk of code that should get you started if you want:
```
const componentsToTest = [
  {
    name: 'anchor',
    tagName: 'a',
    testProps: { href: '...', target: '...' },
    expectedType: 'HTMLAnchorElement',
  },
    ...,
];
```

And then map over the data and write a set of tests like:

```
beforeEach(() => {
  const { getByTestId } = render(<Root {...testProps}>`${name} content`</Root>);
});

test(`it renders as a(n) ${name} element`, () => {
  const el = getByTestId(name);
  expect(el.tagName.toLowerCase()).toBe(tagName);
});

test(`it renders children inside of the ${name}`, () => {
  const el = getByTestId(name);
  expect(el.textContent).toBe(`${name} element`);
});

Object.keys(testProps).forEach(propKey => {
  test(`it correctly sets the ${propKey} element on the ${name} component`, () => {
    const el = getByTestId(name);
    expect((el as expectedType)[propKey]).toBe(testProps[propKey]);
  });
});
```